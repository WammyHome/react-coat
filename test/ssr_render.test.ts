/**
 * @jest-environment node
 */

import server from "./server";

test("ssr success:/photos?search={title:success}", () => {
  const actions: string[] = [];
  const logerMiddleware = ({dispatch}: {dispatch: Function}) => (next: Function) => (originalAction: {type: string}) => {
    actions.push(originalAction.type);
    return next(originalAction);
  };
  expect.assertions(2);
  return server('/photos?search={"title":"success"}', [logerMiddleware]).then(result => {
    // 不验证 connected-react-router 自动加上的一个随机数 key
    delete result.data.router.location.key;
    expect(result).toMatchSnapshot();
    expect(actions).toEqual(["app/INIT", "app/UPDATE", "photos/INIT", "photos/searchList", "photos/UPDATE"]);
  });
});

test("ssr failed:/photos?search={title:error}", () => {
  const actions: string[] = [];
  const logerMiddleware = ({dispatch}: {dispatch: Function}) => (next: Function) => (originalAction: {type: string}) => {
    actions.push(originalAction.type);
    return next(originalAction);
  };
  expect.assertions(2);
  return server('/photos?search={"title":"error"}', [logerMiddleware]).then(result => {
    // 不验证 connected-react-router 自动加上的一个随机数 key
    delete result.data.router.location.key;
    expect(result).toMatchSnapshot();
    expect(actions).toEqual(["app/INIT", "app/UPDATE", "photos/INIT", "photos/searchList", "@@framework/ERROR", "app/putSsrError"]);
  });
});

test("ssr exception:/photos?search={title:exception}", () => {
  const actions: string[] = [];
  const logerMiddleware = ({dispatch}: {dispatch: Function}) => (next: Function) => (originalAction: {type: string}) => {
    actions.push(originalAction.type);
    return next(originalAction);
  };
  expect.assertions(2);
  return server('/photos?search={"title":"exception"}', [logerMiddleware]).catch(e => {
    expect(e.message).toBe("服务器内部错误！");
    expect(actions).toEqual(["app/INIT", "app/UPDATE", "photos/INIT", "photos/searchList", "@@framework/ERROR"]);
  });
});
