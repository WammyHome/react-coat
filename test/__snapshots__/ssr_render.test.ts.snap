// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr failed:/photos?search={title:error} 1`] = `
Object {
  "data": Object {
    "app": Object {
      "curUser": Object {
        "avatarUrl": "/imgs/1.jpg",
        "hasLogin": true,
        "uid": "1",
        "username": "test",
      },
      "isModule": true,
      "loading": Object {
        "global": "Stop",
        "login": "Stop",
      },
      "ssrError": "获取列表失败！",
    },
    "photos": Object {
      "isModule": true,
    },
    "router": Object {
      "action": "POP",
      "location": Object {
        "hash": "",
        "pathname": "/photos",
        "search": "?search={\\"title\\":\\"error\\"}",
        "state": undefined,
      },
    },
  },
  "html": "<div data-reactroot=\\"\\">获取列表失败！</div>",
  "ssrInitStoreKey": "reactCoatInitStore",
}
`;

exports[`ssr success:/photos?search={title:success} 1`] = `
Object {
  "data": Object {
    "app": Object {
      "curUser": Object {
        "avatarUrl": "/imgs/1.jpg",
        "hasLogin": true,
        "uid": "1",
        "username": "test",
      },
      "isModule": true,
      "loading": Object {
        "global": "Stop",
        "login": "Stop",
      },
    },
    "photos": Object {
      "isModule": true,
      "listItems": Array [
        Object {
          "id": "1",
          "title": "item1",
        },
        Object {
          "id": "2",
          "title": "item2",
        },
      ],
      "listSearch": Object {
        "page": 1,
        "pageSize": 10,
        "title": "success",
      },
      "listSummary": Object {
        "page": 1,
        "pageSize": 10,
        "totalItems": 10,
        "totalPages": 1,
      },
    },
    "router": Object {
      "action": "POP",
      "location": Object {
        "hash": "",
        "pathname": "/photos",
        "search": "?search={\\"title\\":\\"success\\"}",
        "state": undefined,
      },
    },
  },
  "html": "<div data-reactroot=\\"\\"><div class=\\"photos\\"><ul><li>photo-<!-- -->item1</li><li>photo-<!-- -->item2</li></ul><div class=\\"pagination\\">共10条，第1/1页</div></div></div>",
  "ssrInitStoreKey": "reactCoatInitStore",
}
`;
